lib/srv/desktop/discovery.go

```diff
@@ -185,19 +186,32 @@ func (s *WindowsService) applyLabelsFromLDAP(entry *ldap.Entry, labels map[strin
 // It checks using the default system resolver first, and falls
 // back to making a DNS query of the configured LDAP server
 // if the system resolver fails.
-func (s *WindowsService) lookupDesktop(ctx context.Context, hostname string) (addrs []string, err error) {
+func (s *WindowsService) lookupDesktop(ctx context.Context, hostname string) ([]string, error) {
 	tctx, cancel := context.WithTimeout(ctx, 10*time.Second)
 	defer cancel()
 
-	addrs, err = net.DefaultResolver.LookupHost(tctx, hostname)
+	stringAddrs := func(addrs []netip.Addr) []string {
+		result := make([]string, 0, len(addrs))
+		for _, addr := range addrs {
+			result = append(result, addr.String())
+		}
+		return result
+	}
+
+	addrs, err := net.DefaultResolver.LookupNetIP(tctx, "ip4", hostname)
 	if err == nil && len(addrs) > 0 {
-		return addrs, nil
+		return stringAddrs(addrs), nil
 	}
 	if s.dnsResolver == nil {
 		return nil, trace.NewAggregate(err, trace.Errorf("DNS lookup for %q failed and there's no LDAP server to fallback to", hostname))
 	}
-	s.cfg.Log.WithError(err).Debugf("DNS lookup for %q failed, falling back to LDAP server", hostname)
-	return s.dnsResolver.LookupHost(ctx, hostname)
+	s.cfg.Log.Debugf("DNS lookup for %q failed, falling back to LDAP server", hostname)
+	addrs, err = s.dnsResolver.LookupNetIP(ctx, "ip4", hostname)
```

Comment:
Do you know why we pass the context with the timeout above, but the passed context without the timeout here?

Reply:
I think the idea was if the default resolver times out, then `tctx` would have alreayd expired and this would be guaranteed to fail.

Reply:
Makes sense, do we perhaps want to create another timeout-context here?